src/sfctl/custom_deployment.py

```diff
@@ -0,0 +1,128 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Custom commands for managing Service Fabric Mesh resources"""
+
+from __future__ import print_function
+
+import enum
+import json
+import os
+import shutil
+from knack.util import CLIError
+from sfmergeutility import SFMergeUtility
+
+class ResourceType(enum.Enum):
+    """ Defines the valid yaml resource types
+        which are parseable by CLI
+    """
+    application = 1
+    volume = 2
+    network = 3
+    secret = 4
+    secretValue = 5
+    gateway = 6
+
+def get_resource_type(file_name):
+    """ Gets the resource type form the file name
+    """
+    file_name = os.path.basename(file_name)
+    file_name = file_name.split('_')
+    if len(file_name) < 3:
+        raise CLIError('Invalid resource file name %s' %(file_name))
+    resource_type = file_name[1]
+    resource_type_to_return = None
+    if resource_type == "application":
+        resource_type_to_return = ResourceType.application
+    elif resource_type == "volume":
+        resource_type_to_return = ResourceType.volume
+    elif resource_type == "network":
+        resource_type_to_return = ResourceType.network
+    elif resource_type == "secret":
+        resource_type_to_return = ResourceType.secret
+    elif resource_type == "secretValue":
+        resource_type_to_return = ResourceType.secretValue
+    elif resource_type == "gateway":
+        resource_type_to_return = ResourceType.gateway
+    return resource_type_to_return
+
+def get_resource_name(file_name):
+    """ Gets resource name form the file name
+    """
+    file_name = os.path.basename(file_name)
+    file_name = file_name.split('_')
+    if len(file_name) < 3:
+        raise CLIError('Invalid resource file name %s' %(file_name))
+    file_name_with_extension = file_name[2]
+    resource_name = file_name_with_extension.split('.')
+    return resource_name[0]
+
+def list_files_directory(directory, extension):

```

Comment:
nit: consider naming this to list_file_in_directory or something along those lines